StripeConnect/StripeConnect/Source/Components/ComponentType.swift

```diff
@@ -8,11 +8,32 @@
 import Foundation
 
 /// The name of the embedded component tag in JS ([docs](https://docs.stripe.com/connect/supported-embedded-components))
-enum ComponentType: String, Encodable {
+enum ComponentType: Encodable {
     /// Displays the balance summary, the payout schedule, and a list of payouts for the connected account
     case payouts
     /// The onboarding flow for the account.
-    case onboarding = "account-onboarding"
+    case onboarding(AccountOnboardingViewController.Props)
     /// Show details of a given payment and allow users to manage disputes and perform refunds.
-    case paymentDetails = "payment-details"
+    case paymentDetails
+
+    var name: String {
+        switch self {
+        case .payouts:
+            return "payouts"
+        case .onboarding:
+            return "account-onboarding"
+        case .paymentDetails:
+            return "payment-details"
+        }
+    }
+
+    func encode(to encoder: any Encoder) throws {
+        switch self {
+        case .onboarding(let initParams):
+            try initParams.encode(to: encoder)
+        case .payouts,
+             .paymentDetails:
+            try VoidPayload().encode(to: encoder)
+        }
+    }
```

Comment:
If we don't override the encoder, the JSON has a shape like:
```
{
    "onboarding": {
        "_0": {
            "setFullTermsOfServiceUrl": "https://..."
        }
    }
}
```

Instead, we want it to have the same shape as the associated value, or empty-json if there is no associated value.